blogs/contractionorder.md

```diff
@@ -0,0 +1,338 @@
+$${dims(b) \times dims(d) \times dims(f)} = 2^{w(b) + w(d) + w(f)}$$
+where $w(*)$ for weights of vertices.
+
+### Tree Decomposition and Tree Width
+
+Intuitively, a [**tree decomposition**](https://en.wikipedia.org/wiki/Tree_decomposition) represents the vertices of a given graph $G$ as subtrees of a tree, in such a way that vertices in $G$ are adjacent only when the corresponding subtrees intersect.
+
+The tree decomposition of a graph is a tree whose nodes are subsets of the vertices of the graph, and the following conditions are satisfied:
+
+1. Each vertex of the graph is in at least one node of the tree.
+2. For each edge of the graph, there is a node of the tree containing both vertices of the edge.
+3. Bags containing the same vertex have to be connected in the tree.
+
+All the nodes of the tree are called **tree bags**, and intersection of two bags is called a **separator**.
+The width of a tree decomposition is the size of the largest bag minus one.
+Clearly, one graph can have multiple tree decompositions, and the **tree width** of a graph is the minimum width of all possible tree decompositions, which is called an optimal tree decomposition.
+For graphs with weights on vertices, we define the width as sum of weights of the vertices in the bags minus one.
+
+An example of the optimal tree decomposition is shown in the following figure:
+
+![](/assets/treewidth_figs/treedecomposition_2.png)
+
+where the left graph is the original graph and the right one is the tree decomposition of the graph, and the tree width is 2.
+
+
+To find the optimal tree decomposition of a simple graph, one can use the Bouchitté-Todinca algorithm [^Bouchitté], which is discussed in detail in the blog [Finding the Optimal Tree Decomposition with Minimal Treewidth](/blogs/treewidth/), one can refer to it for more details.
+
+### From Tree Decomposition to Contraction Order
+
+Then how tree decomposition and tree width are related to the contraction order of the tensor network?
+In this sub-section, we will answer the following questions: how to get a contraction order from a tree decomposition? And why the contraction order is optimal if the tree width is minimal?
+
+For the first question, according to the definition of tree decomposition, we can obtain vertices elimination order by traversing the tree decomposition in a bottom-up manner. 
+For each node of the tree decomposition, a vertex can be eliminated if it is in the bag of the node and not in the bags of its parent node, as shown in the figure below:
+
+![](/assets/treewidth_figs/elimination_order2.png)
+
+and the order is $\{\{i, j, l\}, \{k\}, \{m\}, \{n\}\}$, where the last vertex to be eliminated first.
```

Comment:
I think you miss an "is" here:
..., where the last vertex **is** to be eliminated first.

Reply:
The typos are fixed.